spacy/pipeline/entityruler.py

```diff
@@ -1,16 +1,49 @@
+from typing import Optional, Union, List, Dict, Tuple, Iterable, Any
 from collections import defaultdict
+from pathlib import Path
 import srsly
 
-from ..language import component
+from ..language import Language
 from ..errors import Errors
 from ..util import ensure_path, to_disk, from_disk
 from ..tokens import Doc, Span
 from ..matcher import Matcher, PhraseMatcher
 
 DEFAULT_ENT_ID_SEP = "||"
+PatternType = Dict[str, Union[str, List[Dict[str, Any]]]]
+
+
+@Language.factory(
+    "entity_ruler",
+    assigns=["doc.ents", "token.ent_type", "token.ent_iob"],
+    default_config={
+        "phrase_matcher_attr": None,
+        "validation": False,
+        "overwrite_ents": False,
+        "ent_id_sep": DEFAULT_ENT_ID_SEP,
+        "patterns": None,
+    },
+)
+def make_entity_ruler(
+    nlp: Language,
+    name: str = "entity_ruler",
+    phrase_matcher_attr: Optional[Union[int, str]] = None,
+    validation: bool = False,
+    overwrite_ents: bool = False,
+    ent_id_sep: str = DEFAULT_ENT_ID_SEP,
+    patterns: Optional[List[PatternType]] = None,
+):
```

Comment:
Should we remove the default values from `make_entity_ruler` - as we got them in the `default_config`?

Reply:
Yeah, I've been thinking this, too... I agree would be cleaner to not use defaults in our code and code examples to prevent duplication and make it clear that the _config_ is where the defaults come from. I think it's a reasonable best practice.